packages/shared/src/vault/schemas.ts

```diff
@@ -71,3 +74,50 @@ export const executeSwapParamsSchema = z.union([
 ]);
 
 export type ExecuteSwapParams = z.infer<typeof executeSwapParamsSchema>;
+
+const nativeCallParamsSchema = ethereumBase
+  .extend({ destAsset: erc20, message: z.string(), gasAmount: numericString })
+  .strict();
+
+export type NativeCallParams = z.infer<typeof nativeCallParamsSchema>;
+
+const tokenCallParamsSchema = z.union([
+  ethereumBase.extend({
+    srcAsset: z.literal(Assets.FLIP),
+    destAsset: z.union([z.literal(Assets.USDC), z.literal(Assets.ETH)]),
+    message: hexString,
+    gasAmount: numericString,
+  }),
+  ethereumBase.extend({
+    srcAsset: z.literal(Assets.USDC),
+    destAsset: z.union([z.literal(Assets.FLIP), z.literal(Assets.ETH)]),
+    message: hexString,
+    gasAmount: numericString,
+  }),
+]);
+
+export type TokenCallParams = z.infer<typeof tokenCallParamsSchema>;
+
+export const executeCallParamsSchema = z.union([
+  nativeCallParamsSchema,
+  tokenCallParamsSchema,
+]);
+
+export type ExecuteCallParams = z.infer<typeof executeCallParamsSchema>;
+
+export const executeOptionsSchema = z.intersection(
```

Comment:
feels a bit hard/cumbersome to get the validation right this way and the error messages are not too helpful. think its okay for now, but we might want to validate the possible combinations and stuff in code when we get more combinations